diff --git a/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php b/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php
index 79b715a..8cbb337 100644
--- a/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php
+++ b/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php
@@ -22,6 +22,7 @@
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Security\Core\Role\SwitchUserRole;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
 use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
 use Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
@@ -162,7 +163,7 @@ private function attemptExitUser(Request $request)
             throw new AuthenticationCredentialsNotFoundException('Could not find original Token object.');
         }

-        if (null !== $this->dispatcher) {
+        if (null !== $this->dispatcher && !$original instanceof AnonymousToken) {
             $user = $this->provider->refreshUser($original->getUser());
             $switchEvent = new SwitchUserEvent($request, $user);
             $this->dispatcher->dispatch(SecurityEvents::SWITCH_USER, $switchEvent);
